### PR TITLE
Test elemental strings

### PR DESCRIPTION
**Pull request by rouson**
This pull request adds a [unit test] for the [string conversion utility] and makes each procedure, makes each procedure in the string utility elemental, and converts real variable/constant declarations to use a specified, standard kind.

[unit test]: https://github.com/rouson/icar/blob/test-elemental-strings/tests/test_string.f90
[string conversion utility]: https://github.com/gutmann/icar/blob/master/utilities/string.f90

### CODE REVIEW

**Comment by rouson** [comment]
@gutmann, string.f90 uses what I believe is the most natural way to accomplish your goal.  In the process of looking at whether there is anything I'd change, I decided to write a test so that could have confidence in my changes.  I don't believe I changed any statements inside your procedures.  I just made the procedures elemental to broaden their usefulness and to make clear to the compiler and reader that the procedures are (implicitly) pure.  

It would be great to see these procedures make it into a library that aims for broad use.  Would you be interested in contributing the string.f90 and test_string.f90 to the [functional-fortran] project?

[functional-fortran]: https://github.com/wavebitscientific/functional-fortran

**Comment by gutmann** [comment]
Thanks, these all look great, one question (your going to hate this)

Will this compile with gfortran<4.9? (I can probably live with it if it needs to be more recent, but it will involve some pretty substantial changes to the CI setup).  

Can you submit this pull request against the NCAR/icar repository develop branch? 

I'll have to think about whether it makes sense to use iso_fortran_env to define real types.  There are legacy reasons for not defining them explicitly elsewhere in the code, but that doesn't mean they couldn't be so defined here, and certainly for contribution to a broader audience, that is the way it should be done. 

Yes, I think contributing this to functional-fortran would be great.  I'm sort of surprised they don't have something like this already.

File: tests/test_string.f90 [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  ! Construct expected results
+  character(len=32) :: real64_string, real32_string, integer_string
+
+  write(real64_string,*) real64_datum
+  write(real32_string,*) real32_datum
+  write(integer_string,*) integer_datum
+
+  ! Test each string conversion function
+  if ( get_double(real64_string)   == real64_datum         ) test_passed(get_double_) = .true.
```
Comment (gutmann) [inline tests/test_string.f90]: I assume these tests pass at the moment, but are there potential issues with precision?  Should it only check that the difference between values is less than some small value rather then requiring them to be exactly equal?

**Comment by rouson** [comment]
Yes, the tests currently pass. The tests pass on macOS with gfortran 4.9.4, 5.4.0, 6.3.0, and 7.0.1.  I just tried using MacPorts to build 4.8, but it failed to build.  As you probably know, versions < 4.9 are four or more years old and versions < 5 are unsupported so there won't even be new bug-fix releases for versions < 5.

Your suggested change is reasonable and I'd be glad to push a new commit that uses a tolerance instead of exact equality.  I guess I was thinking that this could be one of the rare instances in which exact equality is a reasonable expectation, but I'm not certain about that and don't have a strong feeling either way.

**Pull request merged**